scripts/data/src/org/akvo/gae/remoteapi/DataStoreWithUnilog.java

```diff
@@ -0,0 +1,265 @@
+package org.akvo.gae.remoteapi;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.akvo.flow.events.EventUtils;
+
+import com.google.appengine.api.datastore.DatastoreAttributes;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Index;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyRange;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Transaction;
+import com.google.appengine.api.datastore.TransactionOptions;
+
+import static org.akvo.flow.events.EventUtils.*;
+
+public class DataStoreWithUnilog implements DatastoreService {
+    private static final Date MANUAL_FIX_START_TIME = new Date();
+    private DatastoreService delegate;
+    private String orgId;
+    private boolean newEventsPushedToEventQueue;
+
+    DataStoreWithUnilog(DatastoreService delegate, String orgId) {
+        this.delegate = delegate;
+        this.orgId = orgId;
+    }
+
+    @Override
+    public Key put(Entity entity) {
+        Key put = delegate.put(entity);
+        storeEntityUpdate(Stream.of(entity));
+        return put;
+    }
+
+    @Override
+    public Key put(Transaction transaction, Entity entity) {
+        Key put = delegate.put(transaction, entity);
```

Comment:
Minor comment: its a funny name for the variable. 😄

Reply:
Intellij default name! Didn't even notice :P. Fixed